bvm/ballerina-core/src/main/java/org/ballerinalang/model/values/BValueArray.java

```diff
@@ -0,0 +1,573 @@
+                BValueArray byteArray = new BValueArray(Arrays.copyOf(byteValues, byteValues.length));
+                byteArray.size = this.size;
+                refs.put(this, byteArray);
+                return byteArray;
+            } else if (elementType.getTag() == TypeTags.FLOAT_TAG) {
+                BValueArray floatArray = new BValueArray(Arrays.copyOf(floatValues, floatValues.length));
+                floatArray.size = size;
+                refs.put(this, floatArray);
+                return floatArray;
+            } else if (elementType.getTag() == TypeTags.STRING_TAG) {
+                BValueArray stringArray = new BValueArray(Arrays.copyOf(stringValues, stringValues.length));
+                stringArray.size = this.size;
+                refs.put(this, stringArray);
+                return stringArray;
+            }
+        }
+
+        BRefType<?>[] values = new BRefType[size];
+        BValueArray refValueArray = new BValueArray(values, arrayType);
+        refValueArray.size = this.size;
+        refs.put(this, refValueArray);
+        int bound = this.size;
+        IntStream.range(0, bound)
+                .forEach(i -> values[i] = this.refValues[i] == null ? null :
+                        (BRefType<?>) this.refValues[i].copy(refs));
+        return refValueArray;
+
+    }
+
+    @Override
+    public String stringValue() {
+        if (elementType != null) {
+            if (elementType.getTag() == TypeTags.INT_TAG) {
+                StringJoiner sj = new StringJoiner(", ", "[", "]");
+                for (int i = 0; i < size; i++) {
+                    sj.add(Long.toString(intValues[i]));
+                }
+                return sj.toString();
+            } else if (elementType.getTag() == TypeTags.BOOLEAN_TAG) {
+                StringJoiner sj = new StringJoiner(", ", "[", "]");
+                for (int i = 0; i < size; i++) {
+                    sj.add(Boolean.toString(booleanValues[i] == 1));
+                }
+                return sj.toString();
+            } else if (elementType.getTag() == TypeTags.BYTE_TAG) {
+                StringJoiner sj = new StringJoiner(", ", "[", "]");
+                for (int i = 0; i < size; i++) {
+                    sj.add(Integer.toString(Byte.toUnsignedInt(byteValues[i])));
+                }
+                return sj.toString();
+            } else if (elementType.getTag() == TypeTags.FLOAT_TAG) {
+                StringJoiner sj = new StringJoiner(", ", "[", "]");
+                for (int i = 0; i < size; i++) {
+                    sj.add(Double.toString(floatValues[i]));
+                }
+                return sj.toString();
+            } else if (elementType.getTag() == TypeTags.STRING_TAG) {
+                StringJoiner sj = new StringJoiner(", ", "[", "]");
+                for (int i = 0; i < size; i++) {
+                    sj.add("\"" + stringValues[i] + "\"");
+                }
+                return sj.toString();
```

Comment:
can move this logic to a util method, since it common for all cases

Reply:
It seems we cannot. We'll not get any much advantage by doing that because of the way of creating the string is different for each type of arrays. But, we could move a few lines out of the if block.